paimon-core/src/main/java/org/apache/paimon/tag/TagBatchCreation.java

```diff
@@ -0,0 +1,124 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.tag;
+
+import org.apache.paimon.CoreOptions;
+import org.apache.paimon.Snapshot;
+import org.apache.paimon.operation.TagDeletion;
+import org.apache.paimon.table.FileStoreTable;
+import org.apache.paimon.utils.SnapshotManager;
+import org.apache.paimon.utils.TagManager;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+
+/** Tag creation for batch mode. */
+public class TagBatchCreation {
+
+    private static final Logger LOG = LoggerFactory.getLogger(TagBatchCreation.class);
+
+    private static final String BATCH_WRITE_TAG_PREFIX = "batch-write-";
+    private final FileStoreTable table;
+    private final CoreOptions options;
+    private final TagManager tagManager;
+    private final SnapshotManager snapshotManager;
+    private final TagDeletion tagDeletion;
+
+    public TagBatchCreation(FileStoreTable table) {
+        this.table = table;
+        this.snapshotManager = table.snapshotManager();
+        this.tagManager = table.tagManager();
+        this.tagDeletion = table.store().newTagDeletion();
+        this.options = table.coreOptions();
+    }
+
+    public void createTag() {
+        Snapshot snapshot = snapshotManager.latestSnapshot();
+        if (snapshot == null) {
+            return;
+        }
+        Instant instant = Instant.ofEpochMilli(snapshot.timeMillis());
+        LocalDateTime localDateTime = LocalDateTime.ofInstant(instant, ZoneId.systemDefault());
+        String tagName =
+                options.tagBatchCustomizedName() != null
+                        ? options.tagBatchCustomizedName()
+                        : BATCH_WRITE_TAG_PREFIX
+                                + localDateTime.format(DateTimeFormatter.ofPattern("yyyy-MM-dd"));
+        try {
+            // If the tag already exists, delete the tag
+            tagManager.deleteTag(
+                    tagName, tagDeletion, snapshotManager, table.store().createTagCallbacks());
+            // Create a new tag
+            tagManager.createTag(
+                    snapshot,
+                    tagName,
+                    table.coreOptions().tagDefaultTimeRetained(),
+                    table.store().createTagCallbacks(),
+                    false);
+        } catch (Exception e) {
+            LOG.warn("Failed to create tag", e);
```

Comment:
Suggestion : `Failed to create tag '${tagname}' from '${snapshot}', you can create this tag manually.`